source/codegen/metadata/nidaqmx/functions.py

```diff
@@ -7283,14 +7282,133 @@
             },
             {
                 'callback_params': [
+                    {
+                        'direction': 'out',
+                        'include_in_proto': False,
+                        'name': 'task',
+                        'type': 'TaskHandle'
+                    },
+                    {
+                        'direction': 'out',
+                        'name': 'status',
+                        'type': 'int32'
+                    }
                 ],
                 'direction': 'in',
                 'include_in_proto': False,
                 'name': 'callbackFunction',
                 'type': 'DAQmxDoneEventCallbackPtr'
             },
             {
-                'direction': 'out',
+                'callback_token': True,
+                'direction': 'in',
+                'include_in_proto': False,
+                'name': 'callbackData',
+                'type': 'void'
+            }
+        ],
+        'returns': 'int32',
+        'stream_response': True
+    },
+    'RegisterEveryNSamplesEvent': {
+        'parameters': [
+            {
+                'direction': 'in',
+                'name': 'task',
+                'type': 'TaskHandle'
+            },
+            {
+                'direction': 'in',
+                'enum': 'EveryNSamplesEventType',
+                'name': 'everyNSamplesEventType',
+                'type': 'int32'
+            },
+            {
+                'direction': 'in',
+                'name': 'nSamples',
+                'type': 'uInt32'
+            },
+            {
+                'direction': 'in',
+                'name': 'options',
+                'type': 'uInt32'
```

Comment:
Should we make this an enum? (Does gRPC support bitflag-style enums?)

Reply:
Great question!

`protobuf` doesn't support bitflags exactly. My general solution for bitflags is (a) make it an enum, (b) if you want to set multiple flags you can use the `set_X_raw` method and | in the 2 enum values (in most languages!) .

[Example](https://github.com/ni/grpc-device/blob/3d674d75127842a08e0d5bbaf991e1e755e97a6b/source/tests/system/nidaqmx_driver_api_tests.cpp#L535)

This only has one flag and 2 possible values, so even the caveats aren't really an issue.

**BUT** the flag is to allow calling the callback on the registering thread, which requires pumping messages on the registering thread. I don't have a way to implement that in our grpc server and I don't think we should mess with it. SO the fix for this is actually to mark them `include_in_proto: False` and hardcode the value to zero. I have that tracked as AB#{1560237}.